kore/test/Test/Kore/Step.hs

```diff
@@ -298,6 +301,33 @@ test_stepStrategy =
             Right _ ->
                 assertFailure "Expected exception LimitExceeded"
 
+hasRewrite :: Strategy Prim -> Bool
+hasRewrite = \case
+    Strategy.Seq s1 s2 -> hasRewrite s1 || hasRewrite s2
+    Strategy.And s1 s2 -> hasRewrite s1 || hasRewrite s2
+    Strategy.Or s1 s2 -> hasRewrite s1 || hasRewrite s2
+    Strategy.Apply Rewrite -> True
+    Strategy.Apply _ -> False
+    Strategy.Stuck -> False
+    Strategy.Continue -> False
+
+prop_alwaysRewrite :: Natural -> Bool
+prop_alwaysRewrite depthLimit =
+    all hasRewrite $ takeWithin (Limit depthLimit) (toList executionStrategy)
+
+getPrims :: Strategy Prim -> [Prim]
+getPrims = \case
+    Strategy.Seq s1 s2 -> getPrims s1 ++ getPrims s2
+    Strategy.Apply p -> [p]
+    Strategy.Continue -> []
+    _ -> [] -- ?
```

Comment:
Let's write a function `isLastSimplify :: Strategy Prim -> Bool` in the same spirit as `hasRewrite`, so that we can cover all the cases better.